### ISSUE TITLE
TypeScript navigate() function missing { route, params } option

### ISSUE THREAD

**Issue by lukewlms** [react-navigation/react-navigation#7676]
**Current Behavior**

In order to call the version of navigate() that allows me to specify a `route` and `params`, I have to type the option param as `any`.  I can get the call to work as in the docs, but I can't find a way to make TypeScript like it.

```ts
    navigator.navigate(TrackingStackRoute.TreatStackNavigator, {
      screen: "TreatListScreen",
      params: { dateStr },
    } as any);
```

**Expected Behavior**

`as any` should be unnecessary here; ideally we would be able to pass `screen` and `params,` and also get type checking on the params:

```ts
    navigator.navigate(TrackingStackRoute.TreatStackNavigator, {
      screen: "TreatListScreen",
      params: { dateStr },
    });
```

Here's the type info I see for `navigate()` currently:

```ts
    /**
     * Navigate to a route in current navigation tree.
     *
     * @param name Name of the route to navigate to.
     * @param [params] Params object for the route.
     */
    navigate<RouteName extends keyof ParamList>(...args: ParamList[RouteName] extends undefined | any ? [RouteName] | [RouteName, ParamList[RouteName]] : [RouteName, ParamList[RouteName]]): void;
    /**
     * Navigate to a route in current navigation tree.
     *
     * @param route Object with `key` or `name` for the route to navigate to, and a `params` object.
     */
    navigate<RouteName extends keyof ParamList>(route: {
        key: string;
        params?: ParamList[RouteName];
    } | {
        name: RouteName;
        key?: string;
        params: ParamList[RouteName];
    }): void;
```

**Your Environment**

RN 61.5
"@react-navigation/native": "^5.0.8",

**Comment by gibo77** [comment]
I used this link below to pass parameters on different pages and so far it has been fantastic. When I followed the docs, it does not work for me. The way the programming of `function` instead of `class` is way off the React Native way of programming should I put it. I get errors if I follow the docs. Hopefully they will align their codes from the creators of React Native.
[](url)https://github.com/JulianCurrie/CwC_React_Native/blob/react_navigation_v5/App.js
